nemo/collections/tts/models/magpietts.py

```diff
@@ -62,6 +63,90 @@ def worker_init_fn(worker_id):
     dataset.text_tokenizer = tokenizer
     dataset.text_conditioning_tokenizer = text_conditioning_tokenizer
 
+def build_vocabs(subword_vocab_items):
+    # tokenizer = AutoTokenizer.from_pretrained(pretrained_tokenizer_name)
+    # subword_vocab_items = tokenizer.vocab.items()
+    org_char_vocab = {subword: subword_id for subword, subword_id in subword_vocab_items if len(subword) == 1}
+    sorted_char_vocab = dict(sorted(org_char_vocab.items(), key=lambda x: x[1]))
+    char_vocab = {k: i for i, (k, _) in enumerate(sorted_char_vocab.items())}
+    assert sorted(char_vocab.values()) == list(range(len(char_vocab)))
+    subword_id_to_char_ids = {
+        subword_id: tuple(char_vocab[char] for char in subword) for subword, subword_id in subword_vocab_items
+    }
+
+    assert max(subword_id_to_char_ids) == len(subword_id_to_char_ids) - 1
+    # add padding_idx
+    subword_padding_idx = len(subword_id_to_char_ids)
+    subword_id_to_char_ids[subword_padding_idx] = (len(char_vocab),)
+
+    return subword_id_to_char_ids, char_vocab, subword_padding_idx
+
+
+def sequence_mask(lengths: Tensor, max_length: int | None = None) -> Tensor:
+    if max_length is None:
+        max_length = int(lengths.max().item())
+    x = torch.arange(max_length, dtype=lengths.dtype, device=lengths.device)
+    return x.unsqueeze(0) < lengths.unsqueeze(1)
+
+class CharAwareSubwordEncoder(NeuralModule):
```

Comment:
Can you move this to https://github.com/NVIDIA/NeMo/blob/magpietts_2503/nemo/collections/tts/modules/magpietts_modules.py?
And add a docstring

Reply:
Done.